powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/enums/PowerAuthSignatureTypes.java

```diff
@@ -77,12 +77,14 @@
      * @return Enum value.
      */
     public static PowerAuthSignatureTypes getEnumFromString(String value) {
-        PowerAuthSignatureTypes type = map.get(value.toLowerCase());
-        if (type == null) { // try to guess the most usual suspect...
-            return PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE;
-        } else {
-            return type;
+        if (value != null) {
+            final PowerAuthSignatureTypes type = map.get(value.toLowerCase());
+            if (type != null) {
+                return type;
+            }
         }
+        // Otherwise try to guess the most usual suspect...
+        return PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE;
```

Comment:
I am not sure if this is a good idea. This way we will allow mistyped signature types which will magically work as `POSSESSION_KNOWLEDGE`. It is better to throw an exception IMO.

Reply:
I agree with you, but it's an old code. I would change this in a separate refactoring, because adding an exception here will break many other things (in restful integration for example).

Reply:
Ok, let's raise a ticket.